compiler/noirc_frontend/src/tests.rs

```diff
@@ -1523,6 +1541,25 @@ fn numeric_generic_as_struct_field_type_fails() {
     check_errors(src);
 }
 
+#[test]
+fn type_alias_to_numeric_generic() {
+    let src = r#"
+    type Double<let N: u32>: u32 = N * 2;
+    fn main() {
+        let b: [u32; 6] = foo();
+        assert(b[0] == 0);
+    }
+    fn foo<let N:u32>() -> [u32;Double::<N>] {
+        let mut a = [0;Double::<N>];
+        for i in 0..Double::<N> {
+            a[i] = i;
+        }
+        a
+    }
+    "#;
+    assert_no_errors(src);
+}
```

Comment:
Can we test composing these? 

```noir
type Quadruple<let N; u32>: u32 = Double<Double<N>>;
```

Reply:
I could not make it work in a loop bound, so I disallow it. It can be worked out in another PR.